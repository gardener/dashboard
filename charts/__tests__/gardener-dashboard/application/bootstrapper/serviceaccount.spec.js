//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { merge } = require('lodash')

const { helm } = fixtures

const renderTemplates = helm.renderBootstrapperApplicationTemplates

describe('gardener-dashboard-terminal-bootstrapper', () => {
  describe('serviceaccount', () => {
    let values
    let templates

    beforeEach(() => {
      values = {
        global: {
          bootstrapper: {
            enabled: true
          }
        }
      }
      templates = [
        'serviceaccount'
      ]
    })

    it('should not render the template', async () => {
      values.global.bootstrapper.enabled = false
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [serviceaccount] = documents
      expect(serviceaccount).toBe(null)
    })

    it('should render the template', async () => {
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [serviceaccount] = documents
      expect(serviceaccount).toMatchSnapshot()
    })

    describe('when virtual garden is enabled', () => {
      beforeEach(() => {
        merge(values.global, {
          virtualGarden: {
            enabled: true,
            terminalBootstrapperUserName: 'runtime-cluster:system:serviceaccount:garden:gardener-dashboard-terminal-bootstrapper'
          }
        })
      })

      it('should not render the template', async () => {
        const documents = await renderTemplates(templates, values)
        expect(documents).toHaveLength(1)
        const [serviceaccount] = documents
        expect(serviceaccount).toBe(null)
      })
    })
  })
})
