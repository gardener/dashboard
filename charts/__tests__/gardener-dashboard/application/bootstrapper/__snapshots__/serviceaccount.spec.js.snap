// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard-terminal-bootstrapper serviceaccount should render the template 1`] = `
Object {
  "apiVersion": "v1",
  "kind": "ServiceAccount",
  "metadata": Object {
    "labels": Object {
      "app": "gardener-dashboard",
      "chart": "application-0.1.0",
      "heritage": "Helm",
      "release": "application",
      "role": "terminal-bootstrapper",
    },
    "name": "gardener-dashboard-terminal-bootstrapper",
    "namespace": "garden",
  },
}
`;
